aexpect/client.py

```diff
@@ -123,9 +124,9 @@ def __init__(self, command=None, a_id=None, auto_close=False, echo=False,
         self.a_id = a_id or data_factory.generate_random_string(8)
         self.log_file = None
         self.closed = False
-        # Use PYTHONENCODING or utf-8 (instead of ascii)
-        self.encoding = os.environ.get("PYTHONENCODING", "utf-8")
-
+        self.encoding = locale.getpreferredencoding()
```

Comment:
Although I like the use of undeniably better `lang.getpreferredencoding()` I don't like the fact that this code still attempts to be smarter than python developers. I tried to break `getpreferredencoding` it resulted in ASCII and in sources I saw they do bunch of checks resulting usually in returning `sys.flags.utf8_mode` (which is not in my fedora-based python). Would you have anything against using it without the custom override?

Although I think it'd be useful to being able to force it on `__init__` as aexpect can be used on remote machines. Theoretically they can be running it on ASCII-only system, logging-in a utf-8 system which can immediately break the drainers...

Reply:
I mean the scenario is:

```bash
LC_ALL=C python -c 'import aexpect; aexpect.ShellSession("socat /dev/ttyS1 -")'
```